src/finalisers/umd.ts

```diff
@@ -96,7 +96,7 @@ export default function umd(
 	}
 
 	const wrapperIntro = `(function (global, factory) {
-			typeof exports === 'object' && typeof module !== 'undefined' ? ${cjsExport}factory(${cjsDeps.join(
+			typeof exports === 'object' && typeof module !== 'undefined' && !module.nodeType ? ${cjsExport}factory(${cjsDeps.join(
```

Comment:
An alternative here might just be `typeof module !== 'undefined' && typeof module.exports === 'object'`... better detection of a feature is always preferable to ruling out negative cases.

Reply:
Your suggestion is awesome. After searching a while, I found that UMD proposal has provided us a shorter [choice](https://github.com/umdjs/umd/blob/master/templates/returnExports.js#L21):

```
typeof module === 'object' && module.exports
```